jjb/jobs/qcs-container-install.yaml

```diff
@@ -33,13 +33,9 @@
                 stable: true
         - config-file-provider:
             files:
-              - file-id: '0f157b1a-7068-4c75-a672-3b1b90f97ddd'
-                target: '${{WORKSPACE}}/rhel7-rcm-internal.repo'
-        - shining-panda:
-            build-environment: virtualenv
-            python-version: System-CPython-2
```

Comment:
I did some investigation and found this:

```
# shell build step
+ which python
/usr/bin/python
+ python --version
Python 2.6.6
+ which pip
/usr/bin/pip
+ pip --version
pip 7.1.0 from /usr/lib/python2.6/site-packages (python 2.6)


# Virtualenv build step
[elyezer-test-rhel6-python] $ /usr/bin/python2.6 /home/jenkins/shiningpanda/jobs/bd110714/virtualenv.py /home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9
New python executable in /home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/python2.6
Also creating executable in /home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/python
Installing setuptools, pip, wheel...done.
[elyezer-test-rhel6-python] $ /bin/sh -xe /tmp/shiningpanda7236294806229336583.sh
+ which python
/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/python
+ which pip
/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/pip
+ cat /home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/pip
#!/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/python2.6

# -*- coding: utf-8 -*-
import re
import sys

from pip._internal import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
+ python --version
Python 2.6.6
+ pip --version
Traceback (most recent call last):
  File "/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/bin/pip", line 7, in <module>
    from pip._internal import main
  File "/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/lib/python2.6/site-packages/pip/_internal/__init__.py", line 42, in <module>
    from pip._internal import cmdoptions
  File "/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/lib/python2.6/site-packages/pip/_internal/cmdoptions.py", line 16, in <module>
    from pip._internal.index import (
  File "/home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/lib/python2.6/site-packages/pip/_internal/index.py", line 526
    {str(c.version) for c in all_candidates},
                      ^
SyntaxError: invalid syntax
```

What is interesting is that from the shell the proper pip is being used. For some reason a different pip is being used when running from within the virtualenv build step.

Reply:
I tried to cat the contents of pip's __init__.py and got this

```
+ cat /home/jenkins/shiningpanda/jobs/bd110714/virtualenvs/d41d8cd9/lib/python2.6/site-packages/pip/__init__.py
__version__ = "10.0.0b2"
```

I wonder from where this version is being picked

Reply:
So the version 10.0.0.b2 is the latest available on PyPI. Version 9.x has Python 2.6 compatibility.

I mean I wonder why virtualenv build step is not picking a proper version for a python2.6

Reply:
Yeah I don't know. I don't know if it is a bug in shining panda or what. :scream: 

I guess my stance is I don't see what value the virtual environment is adding, so if it is causing us trouble, why use it?

Reply:
`{str(c.version) for c in all_candidates}` is actually a set comprehension:

```python
In [2]: {str(c) for c in [1, 2, 3, 4]}
Out[2]: {'1', '2', '3', '4'}

In [3]: type({str(c) for c in [1, 2, 3, 4]})
Out[3]: set
```